src/components/app-bar/AppBar.tsx

```diff
@@ -2,12 +2,43 @@ import React from 'react';
 import {StyleSheet, Text, View} from 'react-native';
 import Icon from 'react-native-vector-icons/MaterialIcons';
 
-export const AppBar: React.FC = () => {
+interface AppBarProps {
+  title?: string;
+  showMenuButton?: boolean;
+  showFavButton?: boolean;
+  onFavPress?: () => void;
+}
+
+export const AppBar: React.FC<AppBarProps> = ({
+  title = '',
+  showMenuButton = false,
+  showFavButton = false,
+  onFavPress,
+}) => {
   return (
     <View style={styles.barContainer}>
-      <Icon name={'menu'} size={25} color="#fff" />
-      <Text style={styles.title}>Ecommerce Store</Text>
-      <Icon name={'shopping-cart'} size={25} color="#fff" />
+      {showMenuButton ? (
+        <Icon name={'menu'} size={25} color="#fff" />
+      ) : (
+        <Icon name={'arrow-back'} size={25} color="#fff" />
+      )}
+      <Text style={styles.title}>{title}</Text>
+      <View style={styles.icons}>
+        {showFavButton && (
+          <Icon
+            onPress={onFavPress}
+            name={'favorite-border'}
+            size={25}
+            color="#fff"
```

Comment:
Also I can see that you pass `size` and `color` to all icons. So it can be:
```js
const ICON_SIZE = 25;
const ICON_COLOR = '#fff';
const commonIconProps = { size: ICON_SIZE, color: ICON_COLOR };

<Icon onPress={onFavPress} name={'favorite-border'} {...commonIconProps} />
```